### PR TITLE
shell: add a directory for backends

### PR DESCRIPTION
**Pull request by jakub-uC**
This change organizes the file structure for the shell.
Files implementing backends are moved to a separate folder.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

### CODE REVIEW

File: subsys/shell/CMakeLists.txt [hunk 1]
```diff
@@ -50,11 +31,6 @@ zephyr_sources_ifdef(
   shell_history.c
 )
 
-zephyr_sources_ifdef(
-  CONFIG_SHELL_LOG_BACKEND
```
Comment (nordic-krch) [inline subsys/shell/CMakeLists.txt:54]: i think that shell log backend shall stay in main dir. It's not a shell backend

File: subsys/shell/backends/shell_telnet.c [hunk 2]
```diff
@@ -13,7 +13,7 @@
 #include <net/net_pkt.h>
 #include <shell/shell_telnet.h>
 
-#include "shell_telnet_protocol.h"
+#include "../shell_telnet_protocol.h"
```
Comment (rlubos) [inline subsys/shell/backends/shell_telnet.c:16]: I think you should move corresponding private header as well, it's only used by telnet backend.

**Review by rlubos — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@jakub-uC can you rebase please?

**Comment by jakub-uC** [comment]
@carlescufi  Some tests do not pass while they are not related to my change. What's more, locally the same tests pass correctly.
image

**Pull request merged**